### ISSUE TITLE
KOTOR: Mainmenu model animation issue

### ISSUE THREAD

**Issue by Nostritius** [xoreos/xoreos#283]
I tried to add the model for the main menu into it, but it seems, that the animation is somehow broken. The model can be found in rims/mainmenu.rim as mainmenu.mdl
![Without Animation](https://user-images.githubusercontent.com/17826813/40886776-7ae4e7ba-673e-11e8-9c7d-748e20f724fe.png)
![With Animation](https://user-images.githubusercontent.com/17826813/40886777-7de0e7b6-673e-11e8-8cb5-9bdc9cf850dc.png)

**Comment by seedhartha** [comment]
What animation are you trying to play on it? Does "default" behaves correctly?

**Comment by Nostritius** [comment]
The "default" animation is the only possible animation and it's the one playing on the second picture.

**Comment by DrMcCoy** [comment]
I really wonder why that model is different.

Maybe the coordinates for the animation controllers are swapped, somehow? In NWN, models for the GUI (the GUI widgets) use x/y for left-right/up-down and z for which are on top of each other, while ingame models are oriented as if the camera has a top-down view (i.e. z is the height). Though why it would be the animations that different from the rest of the model, I would know, if that really is the case here.

Or maybe this model is the wrong one, one left over from development, and the real model is something else?

**Comment by DrMcCoy** [comment]
Animations of doors in KotOR/KotOR2 are also broken. Maybe related?

The protocol droid animations aren't broken per se, but they translate the model down into the ground by about 1.5 meters. Likewise, the utility droids float above the ground. Might be related, might be a different issue.

**Comment by DrMcCoy** [comment]
Okay, #291 fixes the droid and door animations, yet that mainmenu.mdl is still broken with it. So it was unrelated :P

**Comment by DrMcCoy** [comment]
If this model really is the only one that's broken, maybe it's the wrong one? Maybe mainmenu.mdl is just a left-over from development, and the game actually loads a different one?

**Comment by seedhartha** [comment]
The animation itself is OK when imported through the NeverBlender. Could be that the bone weights are invalid. Is there a branch somewhere I can experiment with?

**Comment by DrMcCoy** [comment]
Not a branch, but a quick hack would be this (against 3979c113e938105e060d45e961d087b2bafcd581):

```
diff --git a/src/engines/kotor/creature.cpp b/src/engines/kotor/creature.cpp
index ebaba971..559801cc 100644
--- a/src/engines/kotor/creature.cpp
+++ b/src/engines/kotor/creature.cpp
@@ -255,6 +255,10 @@ void Creature::loadPortrait(const Aurora::GFF3Struct &gff) {
 }
 
 void Creature::loadAppearance() {
+	_model.reset(loadModelObject("mainmenu"));
+	_model->addDefaultAnimation("default", 100);
+	return;
+
 	PartModels parts;
 
 	getPartModels(parts);
```

That replaces all creatures ingame with the that model. You'd still need to enter the game proper every time to test something, so it's still a bit of clicking.

**Comment by seedhartha** [comment]
Should be fixed with #295

**Comment by Nostritius** [comment]
I have tested it and it works.
